services/ext/api.go

```diff
@@ -1546,6 +1546,35 @@ func (api *PublicAPI) CreateTokenGatedCommunity() (*protocol.MessengerResponse,
 	return api.service.messenger.CreateTokenGatedCommunity()
 }
 
+// Returns response with AC notification when owner token is received
+func (api *PublicAPI) RegisterOwnerTokenReceivedNotification(communityID string) (*protocol.MessengerResponse, error) {
```

Comment:
I am wondering why do we need it as an API? I guess it is because we don't have defined way to talk from wallet to messenger, right?

Reply:
Yes, however I already talked to Stefan and I will create a task for this. We need some kind of listener on wallet feed in the messenger. It should add AC messages on some concrete wallet actions. Also it should promoteToControlNode when signer is set (by observing setting signer transaction). It should happen w/o the client interaction.